include/psa/crypto_platform.h

```diff
@@ -44,57 +44,91 @@
 #include <stdint.h>
 
+#if ( defined(__ARMCC_VERSION) || defined(_MSC_VER) ) && \
+    !defined(inline) && !defined(__cplusplus)
+#define inline __inline
+#endif
+
 /* Integral type representing a key handle. */
 typedef uint16_t psa_key_handle_t;
 
-/* This implementation distinguishes *application key identifiers*, which
- * are the key identifiers specified by the application, from
- * *key file identifiers*, which are the key identifiers that the library
- * sees internally. The two types can be different if there is a remote
- * call layer between the application and the library which supports
- * multiple client applications that do not have access to each others'
- * keys. The point of having different types is that the key file
- * identifier may encode not only the key identifier specified by the
- * application, but also the the identity of the application.
- *
- * Note that this is an internal concept of the library and the remote
- * call layer. The application itself never sees anything other than
- * #psa_app_key_id_t with its standard definition.
- */
-
-/* The application key identifier is always what the application sees as
- * #psa_key_id_t. */
-typedef uint32_t psa_app_key_id_t;
-
-#if defined(MBEDTLS_PSA_CRYPTO_KEY_FILE_ID_ENCODES_OWNER)
+#if defined(MBEDTLS_PSA_CRYPTO_KEY_EXTENDED_ID)
 
-#if defined(PSA_CRYPTO_SECURE)
 /* Building for the PSA Crypto service on a PSA platform. */
 /* A key owner is a PSA partition identifier. */
 typedef int32_t psa_key_owner_id_t;
-#endif
 
 typedef struct
 {
-    uint32_t key_id;
-    psa_key_owner_id_t owner;
-} psa_key_file_id_t;
-#define PSA_KEY_FILE_GET_KEY_ID( file_id ) ( ( file_id ).key_id )
-
-/* Since crypto.h is used as part of the PSA Cryptography API specification,
- * it must use standard types for things like the argument of psa_open_key().
- * If it wasn't for that constraint, psa_open_key() would take a
- * `psa_key_file_id_t` argument. As a workaround, make `psa_key_id_t` an
- * alias for `psa_key_file_id_t` when building for a multi-client service. */
-typedef psa_key_file_id_t psa_key_id_t;
-#define PSA_KEY_ID_INIT {0, 0}
-
-#else /* !MBEDTLS_PSA_CRYPTO_KEY_FILE_ID_ENCODES_OWNER */
-
-/* By default, a key file identifier is just the application key identifier. */
-typedef psa_app_key_id_t psa_key_file_id_t;
-#define PSA_KEY_FILE_GET_KEY_ID( id ) ( id )
-
-#endif /* !MBEDTLS_PSA_CRYPTO_KEY_FILE_ID_ENCODES_OWNER */
+    psa_key_owner_id_t owner_id;
+    psa_key_id_t owner_key_id;
+} psa_key_extended_id_t;
+
+#define PSA_KEY_ID_T psa_key_extended_id_t
+#define PSA_KEY_ID_INIT { 0, 0 }
+#define PSA_KEY_ID_GET_ID( key_id ) ( ( key_id ).owner_key_id )
+#define PSA_KEY_ID_GET_OWNER_ID( key_id ) ( ( key_id ).owner_id )
+
+/** Utility to initialize an extended key identifier at runtime.
+ *
+ * \param owner_id Identifier of the key owner.
+ * \param key_id   Identifier of the key.
+ */
+static inline psa_key_extended_id_t psa_key_id_init(
```

Comment:
I'm not sure I like this name because `psa_key_id_init(owner_id, key_id)` is only loosely related to `PSA_KEY_ID_INIT` (which takes no argument). But I don't really have a better name to propose either. `psa_key_id_make`?

We don't have a naming convention for functions like this because we don't yet have functions like this, so this is setting a precedent.

Reply:
I've changed to *_make.